ts/features/it-wallet/saga/itwProximitySaga.ts

```diff
@@ -0,0 +1,84 @@
+import { ProximityManager } from "@pagopa/io-react-native-proximity";
+import { SagaIterator } from "redux-saga";
+import { call, put, takeLatest } from "typed-redux-saga/macro";
+import {
+  ProximityManagerStatusEnum,
+  bleIsEnabled,
+  generateQrCode,
+  proximityManagerStatus,
+  startProximityManager,
+  stopProximityManager
+} from "../store/actions/itwProximityActions";
+
+export function* watchItwProximitySaga(): SagaIterator {
+  // Trigger a saga on bleIsEnabled to check if BLE is enabled or not
+  yield* takeLatest(bleIsEnabled.request, checkBleEnablementSaga);
+
+  // Start Proximity Manager
+  yield* takeLatest(
+    startProximityManager.request,
+    handleStartProximityManagerSaga
+  );
+
+  // Get QR code request
+  yield* takeLatest(generateQrCode.request, handleGenerateQrCodeSaga);
+
+  // Stop Proximity Manager
+  yield* takeLatest(
+    stopProximityManager.request,
+    handleStopProximityManagerSaga
+  );
+}
+
+// start proximity manager
+function* handleStartProximityManagerSaga(): SagaIterator {
+  try {
+    yield* call(ProximityManager.start);
+    yield* put(startProximityManager.success(true));
+    yield* put(generateQrCode.request());
+  } catch {
+    yield* put(startProximityManager.failure(new Error("Start failed")));
+  }
+}
+
+// generate qr code
+function* handleGenerateQrCodeSaga(): SagaIterator {
+  try {
+    const qrCode = yield* call(ProximityManager.generateQrCode);
+    yield* put(generateQrCode.success(qrCode));
+    yield* put(
+      proximityManagerStatus({
+        status: ProximityManagerStatusEnum.STARTED
+      })
+    );
+  } catch {
+    yield* put(generateQrCode.failure(new Error("QR code generation failed")));
+  }
+}
+
+// stop proximity manager
+function* handleStopProximityManagerSaga(): SagaIterator {
+  try {
+    // TODO: remove all listners before stopping the proximity manager
+    // we need a new method in the proximity manager [SIW-775]
+    yield* call(ProximityManager.stop);
+    yield* put(stopProximityManager.success(true));
+  } catch {
+    yield* put(stopProximityManager.failure(new Error("Stop failed")));
+  }
+  yield* put(
+    proximityManagerStatus({
+      status: ProximityManagerStatusEnum.STOPPED
+    })
+  );
+}
```

Comment:
Should we put this in the try block? 
Also, we are not using `ItWallerErrors`, is this intended?

Reply:
This action is only meant to track the status of the proximity manager and must be stopped in any case. For ItWalletError I was thinking about a next refactor. I don't know what type of error would be useful to map. Maybe I can use a GENERIC. What do you think?

Reply:
I'm adding a generic proximity error type just now....

Reply:
https://github.com/pagopa/io-app/pull/5360/commits/49d115537d0cfa890fe08d3a845b4f27ccdcd4b1